deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2015 Airbnb, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.airbnb.deeplinkdispatch
+
+import java.util.regex.Pattern
+
+class DeepLinkEntry(
+    val uriTemplate: String,
+    val type: Type,
+    /**
+     * The class where the annotation corresponding to where an instance of DeepLinkEntry is declared.
+     */
+    val activityClass: Class<*>,
+    val method: String? = null
+) {
+  private val prefix: String by lazy(mode = LazyThreadSafetyMode.NONE) {
```

Comment:
it would be worth documenting here why we don't use thread safe (ie for performance reasons and because we don't expect parallel usage)